### ISSUE TITLE
Speed up utxo scanning

### ISSUE THREAD

**Issue by landabaso** [farvault/farvault-lib#17]
Cache old scans.

Learn different approaches from electrum, bluewallet, ...

Discussion: Maybe it is a good idea to bring some kind of stage management (such as redux) to farvault-lib.
Perhaps something that already implements caching mechanisms and so on. RTK-Query helps with caching but does not do data normalization. Normalization is something really nice to have.

**Comment by landabaso** [comment]
esplora is quite fast already although it does not allow batching.
The guys al Bluewallet use batching for electrum (and also trigger queries in parallel).

For the moment we keep electrum using sequential & non-batched queries which is rock solid.
A faster electrum client is left for the future.
